blog-tech/2022-10-20-k3s.md

```diff
@@ -0,0 +1,417 @@
+    `image: bitnami/kube-rbac-proxy:latest`
+   
+   b. Install ***Shifu***:
+
+   ` kubectl apply -f shifu/pkg/k8s/crd/install/shifu_install.yml `
+   
+   c. Lable the worker node of K3s:
+  
+     `kubectl label nodes k3s type=worker`
+
+     d. Try to run the Pod on the specified node, e.g. an `nginx` Pod:
+
+   ```
+   kubectl run nginx --image=nginx -n deviceshifu --overrides='{"spec": { "nodeSelector": { "type": "worker"}}}'
+   ```
+     e. Run `kubectl get pods -n deviceshifu -owide`, we can see that we have successfully run the pod on the edge node `k3s`
+   ```
+   #kubectl get pods -n deviceshifu -owide
+   NAME READY STATUS RESTARTS AGE IP NODE NOMINATED NODE
+   nginx 1/1 Running 0 42s 10.42.1.3 k3s
+   ````
+
+
+6. deploy a digital twin of the Hikvision camera
+  
+   a. Login to [***shifu.cloud***](https://shifu.cloud)
+
+   ***Shifu Cloud*** is a PaaS based on the open source IoT development framework ***Shifu*** developed by Edgenesis, which provides convenient tools for developers to integrate IoT devices by simply filling in basic information. Without ***Shifu Cloud***, developers need to manually fill in the YAML configuration files and then deploy the virtual devices.
+
+   <img src="/blog-221020/登入cloud.png" width="70%" />
+   
+   b. Add device
+
+   <img src="/blog-221020/添加设备.png" width="70%" />
+
+   c. Select Private Protocol --> Select Hikvision in the drop-down menu, then click Next
+
+   <img src="/blog-221020/选择协议.png" width="70%" />
+
+     
+   d. Add basic information about the device, such as device name, manufacturer and model:\
+
+   <img src="/blog-221020/添加基本信息.png" width="70%" />
+   
+   e. Fill in the IP address, RTSP username and password
+
+   <img src="/blog-221020/IP地址.png" width="60%" />
+   
+   f. Click Access and go to Add Interface
+
+   <img src="/blog-221020/前往添加端口.png" width="60%" />
+  
+   
+   g. Finally click Upload and ***Shifu Cloud*** will automatically generate the YAML file for ***deviceShifu*** (the digital twin)
+
+   <img src="/blog-221020/自动生成Yaml文件.png" width="60%" />
+   
+   h. Copy the command and run it on the master node of K3s
+   <img src="/blog-221020/复制命令.png" width="60%" />
+   
+   i. The output should be as follows:
+      ```
+      root@localhost:~# kubectl apply -f https://******.com/*****/******.yaml
+      configmap/deviceshifu-devicehikvision-configmap created
+      service/deviceshifu-devicehikvision-service created
+      deployment.apps/deviceshifu-devicehikvision-deployment created
+      edgedevice.shifu.edgenesis.io/edgedevice-devicehikvision created
+      root@localhost:~# 
+      ```
+
+      - What happens behind the scenes
+      ***Shifu Cloud*** automatically generates the digital twin's YAML files (Edge Device, ConfigMap, Deployment and Service) with some basic information filled in by the user, if you want to go deeper, go to [github](https://github.com/Edgenesis/shifu/tree/main/examples/rtspDeviceShifu) to see the deployment specific files
```

Comment:
should we encapsulate this section inside info? :
```
:::info

Some **content** with _Markdown_ `syntax`. Check [this `api`](#).

:::

```

Reply:
I tried, you can't add an info here, otherwise the whole format would collapse...